taipy/core/_repository/_sql_repository.py

```diff
@@ -11,18 +11,23 @@
 
 import json
 import pathlib
+from datetime import datetime
+from time import sleep
 from typing import Any, Dict, Iterable, List, Optional, Type, Union
 
 from sqlalchemy.dialects import sqlite
 from sqlalchemy.exc import NoResultFound
 
 from .._repository._abstract_repository import _AbstractRepository
+from ..common._utils import _retry_read_entity
 from ..common.typing import Converter, Entity, ModelType
 from ..exceptions import ModelNotFound
 from .db._sql_connection import _SQLConnection
 
 
 class _SQLRepository(_AbstractRepository[ModelType, Entity]):
+    __EXCEPTIONS_TO_RETRY = (ModelNotFound,)
```

Comment:
The modelNotFound exception meaning is not necessary an error happened. 
It could be that a user is trying to get an entity that has just been deleted by another user.

We don't want to rely on this exception to retry.

What we want is to have an exception that indicates that the query cannot be performed correctly. Not that the entity does not exist. Am I correct?

Reply:
Yep that's true. I will create a new Exception for it later.